powersimdata/scenario/create.py

```diff
@@ -372,3 +371,118 @@ def get_base_grid(self):
+
+        self.exported_methods |= {"set_base_profile", "get_base_profile"}
+
+        self.print_existing_study()
+        self.print_available_profile()
+
+        self.set_base_grid()
+        self.set_change_table()
+
+    def print_available_profile(self):
+        """Print available profiles for the grid model"""
+        print("--> Begin: Available profiles")
+        for p in ["demand", "hydro", "solar", "wind"]:
+            possible = self.get_base_profile(p)
+            if len(possible) != 0:
+                print("%s: %s" % (p, " | ".join(possible)))
+        print("<-- End: Available profiles")
+
+    def set_base_grid(self):
+        """Set base grid"""
+        self.base_grid = Grid(self.interconnect, source=self.grid_model)
+
+    def set_change_table(self):
+        """Set change table"""
+        self.change_table = ChangeTable(self.base_grid)
+
+    def get_base_profile(self, kind):
+        """Return available base profiles.
+
+        :param str kind: one of *'demand'*, *'hydro'*, *'solar'*, *'wind'*.
+        :return: (*list*) -- available version for selected profile kind.
+        """
+        return ProfileInput().get_profile_version(self.grid_model, kind)
+
+    def set_base_profile(self, kind, version):
+        """Set base profile.
+
+        :param str kind: one of *'demand'*, *'hydro'*, *'solar'*, *'wind'*.
+        :param str version: base profile version.
+        :raises ValueError: if no profiles are available or version is not available.
+        """
+        possible = self.get_base_profile(kind)
+        if len(possible) == 0:
+            raise ValueError("No %s profile available" % kind)
+        elif version in possible:
+            if kind == "demand":
+                self.demand = version
+            if kind == "hydro":
+                self.hydro = version
+            if kind == "solar":
+                self.solar = version
+            if kind == "wind":
+                self.wind = version
+        else:
+            raise ValueError("Available %s profiles: %s" % (kind, " | ".join(possible)))
+
+
+class FromPyPSA(_Builder):
+    """Build scenario from a PyPSA Network object
+
+    :param str grid model: the grid model
+    :param list interconnect: list of interconnect(s) to build.
+    :param pandas.DataFrame table: scenario list table.
+    :param \\*\\*kwargs: optional parameters used to instantiate a Grid object.
+    """
+
+    def __init__(self, grid_model, interconnect, table, **kwargs):
+        super().__init__(grid_model, interconnect, table)
+
+        self.reduction = None if "reduction" not in kwargs else kwargs["reduction"]
+
+        self.print_existing_study()
+
+        self.set_base_grid()
+        self.set_change_table()
+
+    def set_base_grid(self):
+        """Set base grid"""
+        raise NotImplementedError()
+
+    def set_change_table(self):
+        """Set change table"""
+        raise NotImplementedError()
```

Comment:
We will have to use functions frm #608 to convert the PyPSA `NetWork` to a `Grid` object and to get the base profiles